File: 0792-binary-search/0792-binary-search.ts
function search(nums: number[], target: number): number {
    var start = 0;
    var end = nums.length-1;
    var ans = -1;
    while(start<=end){
        let mid = start+ Math.floor((end-start)/2);
        const val = nums[mid];
        if(val===target){
            ans = mid;
            break;
        } else if(val<target){
            start = mid+1;
        } else {
            end=mid-1;
        }
    }
    return ans;
};

File: 0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int res = -1;
        int start = 0;
        int end = nums.length-1;
        while (start<= end) {
            int mid = start + (end-start)/2;
            if (nums[mid] == target) {
                res = mid;
                break;
            } else if (nums[mid] < target) {
                start = mid+1;
            } else {
                end =mid-1;
            }
        }
        return res;
    }

    
}